src/libs/actions/Wallet.js

```diff
@@ -105,36 +112,34 @@ function activateWallet(currentStep, parameters) {
                 }
 
                 if (currentStep === CONST.WALLET.STEP.ADDITIONAL_DETAILS) {
+                    // Hide the loader
+                    setAdditionalDetailsLoading(false);
+
                     if (response.title === CONST.WALLET.ERROR.MISSING_FIELD) {
-                        // Errors for missing fields are returned from the API as an array of strings so we are converting this to an
-                        // object with field names as keys and boolean for values
-                        const errorFields = _.reduce(response.data.fieldNames, (errors, fieldName) => ({
+                        // Convert array of strings to object with field names as keys and boolean for values (true if error, false if not)
+                        const errorFields = _.has(response, ['data', 'fieldNames']) ? _.reduce(response.data.fieldNames, (errors, fieldName) => ({
                             ...errors,
                             [fieldName]: true,
-                        }), {});
-                        setAdditionalDetailsLoading(false);
+                        }), {}) : {};
                         setAdditionalDetailsErrors(errorFields);
-                        setAdditionalDetailsErrorMessage('');
-                        return;
+                    }
+
+                    if (response.title === CONST.WALLET.ERROR.FULL_SSN_NOT_FOUND) {
+                        setAdditionalDetailsShouldAskForFullSSN(true);
                     }
 
                     const errorTitles = [
+                        CONST.WALLET.ERROR.FULL_SSN_NOT_FOUND,
                         CONST.WALLET.ERROR.IDENTITY_NOT_FOUND,
                         CONST.WALLET.ERROR.INVALID_SSN,
                         CONST.WALLET.ERROR.UNEXPECTED,
                         CONST.WALLET.ERROR.UNABLE_TO_VERIFY,
                     ];
 
                     if (_.contains(errorTitles, response.title)) {
-                        setAdditionalDetailsLoading(false);
                         setAdditionalDetailsErrorMessage(response.message);
-                        setAdditionalDetailsErrors(null);
-                        return;
                     }
 
-                    setAdditionalDetailsLoading(false);
-                    setAdditionalDetailsErrors(null);
```

Comment:
FYI I removed all these because we were already doing it [there](https://github.com/Expensify/App/pull/7804/files#diff-2ed984677d91a13142dfc8ea0debc7002f6f8c8821e7c460e7a2b85075535e64L86).